### ISSUE TITLE
post.xcat.ng doesn't work on centos8.4 diskful installation.

### ISSUE THREAD

**Issue by ibmxianliqi** [xcat2/xcat-core#7467]
When I want to provision centos8.4 diskful node, I used the post.xcat.ng in the compute.centos8.tmpl for node kickstart template as below: 

%pre
{
echo "Running Kickstart Pre-Installation script..."
#INCLUDE:#ENV:XCATROOT#/share/xcat/install/scripts/pre.rhels8#
} &>>/tmp/pre-install.log

%end

%post
mkdir -p /var/log/xcat/
{
cat >> /var/log/xcat/xcat.log << "EOF"
%include /tmp/pre-install.log
EOF
echo "Running Kickstart Post-Installation script..."
#INCLUDE:#ENV:XCATROOT#/share/xcat/install/scripts/post.xcat.ng#
#INCLUDE:#ENV:XCATROOT#/share/xcat/install/scripts/post.rhels8#
} &>>/var/log/xcat/xcat.log

it will result in the node is always installing forever, because there is no update node status action from xcat.post.ng script.

is there someone to know how to deal with this issue?

**Comment by Obihoernchen** [comment]
Is your management node firewall disabled?

**Comment by samveen** [comment]
An instantiated kickstart file containing the node's name will be generated from this template and stored in the /tftpboot or /install folder. Please check it;'s contents to see whether the generated kickstart contains the expected contents.

**Comment by ibmxianliqi** [comment]
> An instantiated kickstart file containing the node's name will be generated from this template and stored in the /tftpboot or /install folder. Please check it;'s contents to see whether the generated kickstart contains the expected contents.

yes, I'm sure the generated script under the /install/autoinstall directory contains the post section code lines, but the code's post install related code line is not executed during diskful installation.

**Comment by ibmxianliqi** [comment]
is there any progress about this issue?

**Comment by viniciusferrao** [comment]
We ran a full diskful install with `post.xcat.ng` on EL8 (Rocky 8.10) in KVM and it went through fine — curl downloaded postscripts, `updateflag.awk` reported status, node came up clean.

While setting up the test we actually hit something similar: the node sat at "installing" forever. Turned out Anaconda was choking on an unresolved `rootpw` template variable in the kickstart — it never got to `%post` at all. Once we fixed the password in xCAT (`chtab key=system passwd.username=root passwd.password=...`), the whole flow worked.

@ibmxianliqi if you're still seeing this, check the generated kickstart at `/install/autoinst/<nodename>` for unresolved `#TABLE:...#` or `#CRYPT:...#` placeholders. You can also set `XCATDEBUGMODE=1` in the site table to get verbose output during `%post`. The curl logic in `post.xcat.ng` itself works correctly.

Couldn't reproduce and there's been no further diagnostic info, so I think this one can be closed. @Obihoernchen what do you think?

**Comment by Obihoernchen** [comment]
Please reopen if you still hit this issue.
